back/lib/cohere_multilingual_embeddings.rb

```diff
@@ -0,0 +1,46 @@
+class CohereMultilingualEmbeddings
+  def initialize(**params)
+    if !params.key? :region # 'eu-central-1'
+      raise 'No AWS region specified for Anthropic Claude model.'
+    end
+
+    @client = Aws::BedrockRuntime::Client.new(params)
+  end
+
+  def self.token_count(str)
+    # From https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed.html:
+    # "For optimal performance, we recommend reducing the length of each text to less than 512 tokens. 1 token is about 4 characters."
+    (str.size / 4.0).ceil
+  end
+
+  def embedding(text, **params)
+    resp = @client.invoke_model invoke_params(text, **params)
+    body_completion resp.body.string
+  end
+
+  protected
+
+  def model_id
+    'cohere.embed-multilingual-v3'
+  end
+
+  def input_type
+    'search_document' # Alternatively 'search_query'
+  end
+
+  private
+
+  def invoke_params(text, **params)
+    json = {
+      'texts' => [text],
+      'input_type' => input_type,
+      'truncate' => 'END'
```

Comment:
This option should be used to truncate the input text if it's too long, but this doesn't seem to work (texts longer than 2048 chars are not accepted).